src/main/java/com/diffplug/gradle/spotless/ApplyFormatTask.java

```diff
@@ -0,0 +1,52 @@
+package com.diffplug.gradle.spotless;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ApplyFormatTask extends DefaultTask {
```

Comment:
This task and `CheckFormat` have quite some duplication, we should extract a base class. This base class can then also be used to get rid of the duplicated configuration logic in the plugin.